src/Offer/Offer.php

```diff
@@ -730,11 +731,9 @@ public function delete(): void
 
     protected function importWorkflowStatus(CultureFeed_Cdb_Item_Base $cdbItem): void
     {
-        try {
-            $workflowStatus = WorkflowStatus::fromNative($cdbItem->getWfStatus());
-        } catch (\InvalidArgumentException $exception) {
-            $workflowStatus = WorkflowStatus::READY_FOR_VALIDATION();
-        }
+        $wfStatus = $cdbItem->getWfStatus();
+        $workflowStatus = $wfStatus ? new WorkflowStatus(str_replace('READYFORVALIDATION', 'READY_FOR_VALIDATION', strtoupper($wfStatus))) : WorkflowStatus::READY_FOR_VALIDATION();
```

Comment:
I'm not sure that this is the best way to handle udb2 imports that use  `readyforvalidation` while udb3 uses `READY_FOR_VALIDATION` but because the switch-off of UDB2 will happen soon. It might doable as a tempfix.

Reply:
Maybe we could make a new static method called `fromCultureFeedWorkflowStatus` inside `WorkflowStatus` and that method contains a mapping from the CultureFeed values to the UDB3 values?

Reply:
I was kinda hesitated to do that, because if the UDB2 import stuff is ever removed from UDB3(which I presume could safely be done, when the switch-off has happened). It would require more refactoring, although I agree that I probably looks cleaner in the `WorkflowStatus` class.

Reply:
Sadly we won't be able to delete all UDB2-related code even after EntryAPI2 and SAPI2 have been switched off. Only the code that process _new_ imports from EntryAPI2, and the CdbXml Service which creates XML for events in UDB3 for SAPI2. But code like this, inside the Event/Place aggregates, needs to keep existing because we will forever have events like `EventImportedFromUdb2` or `PlaceImportedFromUdb2` in our event store which contain CdbXml. So we need to be able to parse that forever (for example here where we use it to determine the state of the event/place, or during replays when we rebuild the read models from the events in the event store). However we don't need to use Culturefeed for that forever, we could do it in a different way (later) to at least remove the dependency on `cultuurnet/culturefeed-php`.

Reply:
Some extra context can be found here https://github.com/cultuurnet/udb3-silex/pull/882/files#r789481893